File: app/src/main/java/com/nurram/project/imagetextrecognition/SavedListActivity.java
package com.nurram.project.imagetextrecognition;

import androidx.lifecycle.Observer;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import android.widget.Toast;

import com.nurram.project.imagetextrecognition.room.Word;
import com.nurram.project.imagetextrecognition.room.WordRepository;
import com.nurram.project.imagetextrecognition.util.PengecualianAdapter;
import com.nurram.project.imagetextrecognition.util.SavedClickUtil;

import java.util.List;

public class SavedListActivity extends AppCompatActivity implements SavedClickUtil {
    private WordRepository repository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_saved_list);

        Toolbar toolbar = findViewById(R.id.pengecualian_toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle("Daftar Pengecualian");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        final RecyclerView view = findViewById(R.id.list_pengecualian);
        view.setLayoutManager(new LinearLayoutManager(this));
        view.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));

        final PengecualianAdapter adapter = new PengecualianAdapter(this, this);
        repository = new WordRepository(getApplication());
        repository.getAllWords().observe(this, new Observer<List<Word>>() {
            @Override
            public void onChanged(@Nullable List<Word> words) {
                adapter.addData(words);
                view.setAdapter(adapter);
            }
        });
    }

    @Override
    public void listClicked(View view, Word words) {
        repository.delete(words);
        Toast.makeText(this, words.getWord() + " deleted", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/nurram/project/imagetextrecognition/SavedListActivity.kt
package com.nurram.project.imagetextrecognition

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.nurram.project.imagetextrecognition.databinding.ActivitySavedListBinding
import com.nurram.project.imagetextrecognition.room.Word
import com.nurram.project.imagetextrecognition.room.WordRepository
import com.nurram.project.imagetextrecognition.util.ExcludeAdapter
import kotlinx.coroutines.DelicateCoroutinesApi

@DelicateCoroutinesApi
class SavedListActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySavedListBinding
    private lateinit var repository: WordRepository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySavedListBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.pengecualianToolbar)
        supportActionBar?.title = getString(R.string.whitelist)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        binding.listPengecualian.layoutManager = LinearLayoutManager(this)
        binding.listPengecualian.addItemDecoration(
            DividerItemDecoration(
            this,
            DividerItemDecoration.VERTICAL)
        )

        val adapter = ExcludeAdapter {
            repository.delete(it)
            Toast.makeText(
                this,
                "${it.word} ${getString(R.string.deleted)}",
                Toast.LENGTH_SHORT)
                .show()
        }
        repository = WordRepository(application)
        repository.getAll().observe(this) { words ->
            adapter.addData(words as ArrayList<Word>)
            binding.listPengecualian.adapter = adapter
        }
    }
}